sdk/core/azure-core-amqp/src/amqp/management.cpp

```diff
@@ -69,9 +69,8 @@ namespace Azure { namespace Core { namespace Amqp { namespace _detail {
     m_eventHandler = nullptr;
     if (m_isOpen)
     {
-      AZURE_ASSERT_MSG(!m_isOpen, "Management being destroyed while open."); // LCOV_EXCL_LINE
-      Azure::Core::_internal::AzureNoReturnPath(
-          "Management is being destroyed while open."); // LCOV_EXCL_LINE
+      AZURE_ASSERT_MSG(!m_isOpen, "Management being destroyed while open.");
+      Azure::Core::_internal::AzureNoReturnPath("Management is being destroyed while open.");
```

Comment:
This is an interesting line when it comes to coverage.  The purest says we need to hit it, but the realist says we should never ever get to it.

Reply:
I literally just added a test case to hit it.

EXPECT_DEATH is your friend. What I *don't* know is if the coverage trace generation is good enough to handle the case where the process terminates. I'm checking that now.

Reply:
Major bummer - code coverage traces didn't hit for Azure::Core::_internal::AzureNoReturnPath. That's a bummer.

But we *do* now have test coverage for it, even though the coverage tools don't think they do.

Reply:
This might be the one legitimate place to have a coverage exclusion along with a note pointing out the test case that covers the scenario.

We should add tooling to scan source files and fail CI if new //LCOV... lines are added.